Added User-Agent header to elasticsearch HTTP(S) transport

g-a-d opened the pull request:
This PR adds a User-Agent header to any HTTP or HTTPS requests.

This is valuable for a number of reasons: 
* analysis of ES request logs
* if a reverse proxy is implemented in front of the ES cluster, allow more intelligent behaviour (e.g. by default redirect to Kibana, but if U-A matches elasticdump then allow access to the REST endpoint)
* more compliant with RFC7231 (which states that we should send a User-Agent)

evantahler commented:
This is a good idea! 
However it fails the test suite, and you have not added a new test which confirms your new behavior.

Please correct the above, and I'll merge it in!

g-a-d commented:
Hi,

Thanks for the quick response to the PR :)
I've added a test and fixed the cause of the original failure (JS standards).

The mocha tests (including the test for this new functionality) now pass on my local machine.

In Travis, though, they appear to be sometimes failing somewhat randomly on tests that, on the surface, appear to have nothing to do with the change made, and more to do with the structure of the data in ES: https://travis-ci.org/taskrabbit/elasticsearch-dump/builds/193852440 and https://travis-ci.org/taskrabbit/elasticsearch-dump/builds/193786147

If you're able to take a look at your convenience that would be great!
Thanks again.

evantahler commented:
Thanks for making the changes!  Sadly, travis.ci can often be unreliable, especially with Elasticsearch.  Re-running the test came back green... 

Merged!

evantahler commented:
v3.1.0 is released with this update
